src/core_ocean/shared/mpas_ocn_vmix_cvmix.F

```diff
@@ -384,8 +384,17 @@ subroutine ocn_vmix_coefs_cvmix_build(meshPool, statePool, forcingPool, diagnost
          cvmix_variables % WaterDensity_cntr => density(1:maxLevelCell(iCell), iCell)
          cvmix_variables % AdiabWaterDensity_cntr => displacedDensity(1:maxLevelCell(iCell), iCell)
 
-         ! fill Ri
          if (kpp_stage == 2) then
+         if (config_cvmix_background_scheme == 'bryanLewis') then
+            do k = 1,maxLevelCell(iCell)
+               vertDiffTopOfCell(k, iCell) = vertDiffTopOfCell(k, iCell) + config_cvmix_bryanLewis_bl1 + &
+                        (config_cvmix_bryanLewis_bl2/pii)*atan((abs(cvmix_variables % zw_iface(k)) -  &
+                        config_cvmix_bryanLewis_transitionDepth) / config_cvmix_bryanLewis_transitionWidth)
```

Comment:
@vanroekel I'm just being thorough - this is equation 9 from the paper, right?
![image](https://user-images.githubusercontent.com/4036816/94285569-0e707780-ff11-11ea-9c33-af71ac479d79.png)
with 
```
  95     config_cvmix_bryanLewis_bl1 = 8.0e-5
  96     config_cvmix_bryanLewis_bl2 = 1.05E-4
  97     config_cvmix_bryanLewis_transitionDepth = 2500
  98     config_cvmix_bryanLewis_transitionWidth = 222.   (1/222. = 0.0045)
```
It all looks right, except he has a 10^4 out front (also in mks units) and you have 10^{-4}.  The paper must have a typo, because he talks about values in the range of 10^{-4}, which makes sense for horizontal mixing.

Also, `abs(cvmix_variables % zw_iface(k))` must be some depth offset.

Reply:
atan is an expensive function. how variable is zw_iface? any chance we could mitigate by precomputing? Understand if we can't. Looks like we could precompute a number of other factors in this expression too.

Reply:
@mark-petersen yes that's the one.

@philipwjones I don't imagine z changing too much, less sure near ice cavities.  I could move that to the init phase of KPP.

Reply:
Looks like `cvmix_variables % zw_iface(k)` is simply depth (z in eqn 9) - @vanroekel is that right?

Over z=0 to 5000, tanh() would have inputs of -10 to 10. It is that same function all the time. In straight z-level we could pre-compute once. We could do some approximation here, like a linear fit at each mid-depth for z-star. It might not apply to more general vertical coordinates, but we don't use those in practice.

@vanroekel is this potentially a default option in E3SM, or just for testing?

Reply:
We're hoping to move to simulations on a mesh with ice-shelf cavities and I don't think it's at all safe to assume z-level anymore.

Reply:
@mark-petersen yes this is depth of interfaces (zTop but referenced to SSH not z=0).  Instead of precomputing, I'd much rather prefer search out a similar, less expensive functional approximation.  Let me look into this

Yes I'd like to target this for default.

Reply:
@vanroekel I think we did a similar substitution for atan in the original KPP. Wish I could remember what we used...

Reply:
@philipwjones exactly what I was thinking.  I found a possible approximation from POP, checking how close it is now.  Will push up a change soon.

Reply:
If you need another option...

From [“Efficient approximations for the arctangent function”, Rajan, S. Sichun Wang Inkol, R. Joyal, A., May 2006](http://www-labs.iro.umontreal.ca/~mignotte/IFT2425/Documents/EfficientApproximationArctgFunction.pdf), `atan` can be approximated via the following C code:
```
#define M_PI_4 (3.1415926535897932384626433832795/4.0)

double FastArcTan(double x) {
  return M_PI_4*x - x*(fabs(x) - 1)*(0.2447 + 0.0663*fabs(x));
}
```

It is valid for `x` values of -1 to 1, a bit over 3x faster than `atan`,  and has a max error of approx 0.0015 radians.

Reply:
@mattdturner this is what I was looking at too, but our range goes from -10, 10, and the error gets very large as soon as you go outside -1 to 1 with this formula

Reply:
@mattdturner and @philipwjones what about something like this?  In fortran of course, this is my python testing
```
for i in range(len(z)):
    x = (abs(z[i])-2500.)/222.
    if abs(x) < 1:
        zmix2[i] = 8e-5 + 1.05E-4/np.pi*(np.pi*x-x*(abs(x)-1)*(0.2447+0.0663*abs(x)))
    else:
        zmix2[i] = 8e-5 + 1.05E-4/np.pi*(np.pi/2*np.sign(x) - 4*x/(4*x**2+1))
```
this gives a nice approximation through the full column

Reply:
thanks to @mattdturner I have a viable alternative
```
zmix4 = 8e-5 + 1.05E-4/2*(np.sign(x)*(0.596227*abs(x)+x**2)/(1+2*0.596227*abs(x)+x**2))
```
This works well through the entire column, without the nasty embedded if.  I'll push this up soon.

Reply:
Great, glad you guys found one. Otherwise I was about to suggest a spline fit where we precompute the spline coefficients and then it becomes just a spline (polynomial) evaluation.  This is better.

Reply:
okay this is changed now.